mod_app/models/support_models.py

```diff
@@ -0,0 +1,126 @@
+from django.db import models
+
+from ckeditor.fields import RichTextField
+
+
+class Tag(models.Model):
+    def __str__(self):
+        return self.name
+
+    name = models.CharField(max_length=255, unique=True)
+
+
+class Link(models.Model):
+    def __str__(self):
+        return self.path
+
+    url = models.URLField()
+    description = models.CharField(
+        max_length=250,
+        help_text="short description of what the link is to (optional)",
+        blank=True,
+        null=True,
+    )
+
+
+class FileLink(Link):
+    def upload_to(instance, filename):
+        return f"files/{instance.related_name}/{filename}"
+
+    file = models.FileField(upload_to=upload_to, blank=True, null=True)
+
+
+class Copy(Link):
+    class Meta:
+        verbose_name_plural = "Copies"
+
+    def __str__(self):
+        if self.name:
+            return self.name
+        else:
+            return super().__str__()
+
+    name = models.CharField(
+        max_length=100,
+        help_text="optional name/title for the copy",
+        blank=True,
+    )
+
+    CONDITION_CHOICES = [
+        ("good", "Good"),
+        ("reasonable", "Reasonable"),
+        ("poor", "Poor"),
+        ("fragile", "Fragile"),
+        ("other", "Other"),
+    ]
+
+    condition = models.CharField(
+        max_length=10,
+        choices=CONDITION_CHOICES,
+        default="reasonable",
+    )
+
+    condition_comments = models.TextField(
+        help_text="Use this to expand on the condition, particularly if 'Other'"
+    )
+
+
+class Analysis(models.Model):
+    class Meta:
+        verbose_name_plural = "Analyses"
+
+    def __str__(self):
+        return f"Analysis of {self.film}"
+
+    content = RichTextField(null=True)
+    film = models.ForeignKey(
+        "Film", on_delete=models.DO_NOTHING, related_name="analyses"
+    )
+
+
+class Actor(models.Model):
+    def __str__(self):
+        return f"{self.name}"
+
+    name = models.CharField(max_length=100)
+    birth_place = models.ForeignKey(
+        "Location", on_delete=models.SET_DEFAULT, default=None, blank=True, null=True
+    )
+    tags = models.ManyToManyField(Tag, related_name="actor_tags")
```

Comment:
This is saying it's required when I try to create an Actor. Should blank=True for this?